httpx/middleware/query_normalizer.go

```diff
@@ -0,0 +1,42 @@
+package middleware
+
+import (
+	"net/http"
+
+	"net/url"
+	"regexp"
+)
+
+// WithQueryNormalizer fixes wrong php query string handling as array
+func WithQueryNormalizer(h http.Handler) http.Handler {
+	rxp, e := regexp.Compile("\\[[0-9]+\\]")
+	if e != nil {
+		panic(e)
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		q := r.URL.Query()
+
+		normalizedQuery := make(url.Values, 0)
```

Comment:
Either use `url.Values{}` or optimise for the expected length. The latter is preferable.